content/acra/configuring-maintaining/general-configuration/acra-server/encryptor-config.md

```diff
@@ -0,0 +1,456 @@
+encryption/masking/tokenizing column's data. AcraServer will use encryption keys related to specified ZoneID even if 
+`--zonemode_enable=false` passed. This option overrides ClientID from TLS certificate or `--client_id` CLI parameter.
+
+#### **crypto_envelope**
+
+Required: `false`
+
+Type: `string`
+
+Values: `acrablock`, `acrastruct`
+
+Group: `common` (doesn't work for `tokenization`)
+
+Description: configures which [crypto envelope](/acra/acra-in-depth/data-structures/) use for transparent encryption or masking. It overrides same option in the 
+`defaults` section.
+
+#### **reencrypting_to_acrablocks**
+
+Required: `false`
+
+Type: `boolean`
+
+Group: `encryption`, `searchable encryption`
+
+Description: turns on re-encryption AcraStructs generated on application side to AcraBlock before passing it to database.
+It overrides same option from `defaults` section.
+
+#### **searchable**
+
+Required: `false`
+
+Type: `boolean`
+
+Group: `searchable encryption`
+
+Description: turns on searchable encryption with [ability to run simple queries](/acra/security-controls/searchable-encryption/) over 
+encrypted data. Final ciphertext will store more data with additional hash value at the beginning of data. Searchable encryption
+supports `acrablock` and `acrastruct` crypto envelopes. This option cannot be used together with masking or tokenization
+related options.
+
+#### **data_type**
+
+Required: `false`
+
+Type: `string`
+
+Values: `str`, `int64`, `int32`, `bytes`
+
+Group: `encryption`, `searchable encryption`, `masking` (`int32`, `int64` not supported for masking)
+
+Description: configures how AcraServer will replace real type of data stored in database with application's type. Due to
+storing data as blobs, AcraServer allow change type on DB protocol level. After that binary data will look like 
+Text/Integer/Binary data types for application. 
+
+How AcraServer maps types from configuration file to DB specific type:
+- `str` - [text](https://www.postgresql.org/docs/current/datatype-character.html) (PostgreSQL, [oid](https://www.postgresql.org/docs/current/datatype-oid.html)=25); [MYSQL_TYPE_STRING](https://dev.mysql.com/doc/internals/en/com-query-response.html#packet-Protocol::ColumnDefinition) (MySQL, 0xfe)
+- `int32` - [integer](https://www.postgresql.org/docs/current/datatype-numeric.html) (PostgreSQL, oid=23), [MYSQL_TYPE_LONG](https://dev.mysql.com/doc/internals/en/com-query-response.html#packet-Protocol::ColumnDefinition) (MySQL, 0x03)
+- `int64` - [bigint](https://www.postgresql.org/docs/current/datatype-numeric.html) (PostgreSQL, oid=20), [MYSQL_TYPE_LONGLONG](https://dev.mysql.com/doc/internals/en/com-query-response.html#packet-Protocol::ColumnDefinition) (MySQL, 0x08)
+- `bytes` - [bytea](https://www.postgresql.org/docs/current/datatype-binary.html) (PostgreSQL, oid=17), [MYSQL_TYPE_BLOB](https://dev.mysql.com/doc/internals/en/com-query-response.html#packet-Protocol::ColumnDefinition) (MySQL, 0xfc)
```

Comment:
Maybe transform it to a table? Something like:

| Data type | PostgreSQL                                                                                                                                    | MySql                                                                                                                          |
| --------- | --------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------ |
| `str`     | [text](https://www.postgresql.org/docs/current/datatype-character.html) ([oid](https://www.postgresql.org/docs/current/datatype-oid.html)=25) | [MYSQL_TYPE_STRING](https://dev.mysql.com/doc/internals/en/com-query-response.html#packet-Protocol::ColumnDefinition) (0xfe)   |
| `int32`   | [integer](https://www.postgresql.org/docs/current/datatype-numeric.html) (oid=23)                                                             | [MYSQL_TYPE_LONG](https://dev.mysql.com/doc/internals/en/com-query-response.html#packet-Protocol::ColumnDefinition) (0x03)     |
| `int64`   | [bigint](https://www.postgresql.org/docs/current/datatype-numeric.html) (oid=20)                                                              | [MYSQL_TYPE_LONGLONG](https://dev.mysql.com/doc/internals/en/com-query-response.html#packet-Protocol::ColumnDefinition) (0x08) |
| `bytes`   | [bytea](https://www.postgresql.org/docs/current/datatype-binary.html) (oid=17)                                                                | [MYSQL_TYPE_BLOB](https://dev.mysql.com/doc/internals/en/com-query-response.html#packet-Protocol::ColumnDefinition) (0xfc)     |

Reply:
transformed. agree, looks prettier